atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaApiTestImpl.java

```diff
@@ -96,4 +105,19 @@ protected void deleteFirstColumn(Transaction transaction, String rowKey) {
         SchemaApiTestTable table = tableFactory.getSchemaApiTestTable(transaction);
         table.deleteColumn1(SchemaApiTestRow.of(rowKey));
     }
+
+    @Test
+    public void testHashFirstTwoRowComponents() {
+        HashComponentsTestTable.HashComponentsTestRow testRow =
+                HashComponentsTestTable.HashComponentsTestRow.of(TEST_VALUE_INTEGER, TEST_VALUE_STRING);
+
+        byte[] component1Bytes = EncodingUtils.encodeUnsignedVarLong(TEST_VALUE_INTEGER);
+        byte[] component2Bytes = EncodingUtils.encodeVarString(TEST_VALUE_STRING);
+        long hashOfFirstTwoComponents =
+                Hashing.murmur3_128().hashBytes(EncodingUtils.add(component1Bytes, component2Bytes)).asLong();
+        byte[] hashOfComponentsBytes = PtBytes.toBytes(Long.MIN_VALUE ^ hashOfFirstTwoComponents);
+
+        assertThat(testRow.persistToBytes())
```

Comment:
also test hydrate to bytes after the persist it should give back testRow. That will be a complete (ser/deser)ialization test.